src/hotspot/share/prims/nativeLookup.cpp

```diff
@@ -272,17 +272,24 @@ address NativeLookup::lookup_style(const methodHandle& method, char* pure_name,
   }
 
   // Otherwise call static method findNative in ClassLoader
+
   Klass*   klass = vmClasses::ClassLoader_klass();
-  Handle name_arg = java_lang_String::create_from_str(jni_name, CHECK_NULL);
+  Handle jni_class(THREAD, method->method_holder()->java_mirror());
```

Comment:
This is the biggest change in this PR. That is, we need to pass enough arguments to `ClassLoader::findNative` so that the method can start a restricted check accordingly.